geopandas/geoseries.py

```diff
@@ -481,6 +483,15 @@ def from_xy(cls, x, y, z=None, index=None, crs=None, **kwargs):
         2    POINT (-3.00000 1.50000)
         dtype: geometry
         """
+        if index is None:
+            is_index_reusable = (
+                isinstance(x, Series)
+                and isinstance(y, Series)
+                and x.index is y.index
+                and (z is None or (isinstance(z, Series) and x.index is z.index))
+            )
+            if is_index_reusable:
+                index = x.index
```

Comment:
```suggestion
            if (
                isinstance(x, Series)
                and isinstance(y, Series)
                and x.index is y.index
                and (z is None or (isinstance(z, Series) and x.index is z.index))
            ):  # check if we can reuse index
                index = x.index
```

Would this be too unreadable? I think we don't need to declare `is_index_reusable`.

Reply:
Probably just a matter of code style preference, let's go with your suggestion for consistency.